uni/lib/controller/fetchers/faculties_fetcher.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:html/parser.dart';
+import 'package:uni/controller/networking/network_router.dart';
+import 'package:uni/model/entities/session.dart';
+
+Future<List<String>> getStudentFaculties(Session session) async {
+  final registerFaculties = <String>[];
+
+  final query = {'pct_codigo': session.username};
+  const baseUrl =
+      'https://sigarra.up.pt/up/pt/vld_entidades_geral.entidade_pagina';
+
+  final pctRequest =
+      await NetworkRouter.getWithCookies(baseUrl, query, session);
+
+  final pctDocument = parse(pctRequest.body);
+
+  final list =
+      pctDocument.querySelectorAll('#conteudoinner>ul a').map((e) => e.text);
+
+  //user is enrolled in only one faculty
+  if (list.isEmpty) {
+    final singleFaculty = pctDocument.querySelector('a')!.attributes['href'];
```

Comment:
From what I can tell if the user still hasn't accepted cookie tracking, this `a` element will be the button to accept the cookies, which has no href. 
![image](https://github.com/NIAEFEUP/uni/assets/13498603/5f0e07f0-f869-44e6-ab74-bf71a1041267)

Reply:
I think a better approach would be to do some URI parsing, on the `link[rel="canonical"]` element. From the URL, you can extract the faculty it belongs to.
This approach could also be applied for the case where the list is not empty (read the href from each anchor and then apply the same method), but I'll leave it to you to choose if you want to do this or not.